services/common/src/main/java/org/finra/gatekeeper/GatekeeperCommonConfig.java

```diff
@@ -160,21 +184,27 @@ public LdapTemplate ldapTemplate(ContextSource contextSource) {
     }
 
     @Bean
-    public GatekeeperAuthorizationService gatekeeperLDAPAuthorizationService(LdapTemplate ldapTemplate,
-                                                                             Supplier<IGatekeeperUserProfile> gatekeeperUserProfileSupplier){
-        //Sets to AD if true
-        if(gatekeeperAuthProperties.getLdap().getIsActiveDirectory()) {
-            logger.info("Setting Authorization to work with Active Directory");
-            return new GatekeeperActiveDirectoryLDAPAuthorizationService(ldapTemplate,
-                    gatekeeperUserProfileSupplier,
-                    gatekeeperAuthProperties);
+    public GatekeeperAuthorizationService gatekeeperAuthorizationService(LdapTemplate ldapTemplate,
+                                                                         Supplier<IGatekeeperLDAPUserProfile> gatekeeperLDAPUserProfileSupplier,
+                                                                         Supplier<IGatekeeperHeaderUserProfile> gatekeeperHeaderUserProfileSupplier){
+        String authType = gatekeeperAuthProperties.getAuthServiceType().toUpperCase();
+        switch (authType){
+            case "HEADERS" :
```

Comment:
I Feel like an Enum would just look so much cleaner when it comes to doing switch/case values